psiminer-cli/src/main/kotlin/Metrics.kt

```diff
@@ -0,0 +1,49 @@
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiJavaCodeReferenceElement
+import com.intellij.psi.PsiJavaFile
+import com.intellij.psi.util.descendantsOfType
+
+interface Metric {
+    val name: String
+    fun calculate(project: Project): Double
+}
+
+abstract class ReferencesMetric : Metric {
+    fun findPercentageOfResolvedReferences(references: Sequence<PsiJavaCodeReferenceElement>): Double {
+        var nReferences = 0
+        var nResolvedReferences = 0
+        for (reference in references) {
+            nReferences += 1
+            if (reference.resolve() != null) {
+                nResolvedReferences += 1
+            }
+        }
+        return if (nReferences > 0) {
+                (nResolvedReferences.toDouble() / nReferences.toDouble()) * 100
+        } else 100.0
+    }
+
+    abstract fun getReferences(psiJavaFile: PsiJavaFile): Sequence<PsiJavaCodeReferenceElement>
+
+    override fun calculate(project: Project): Double {
+        val references = project.psiFiles.asSequence().filterIsInstance<PsiJavaFile>().flatMap { getReferences(it) }
+        return findPercentageOfResolvedReferences(references)
+    }
+}
+
+object ResolvedReferencesMetric : ReferencesMetric() {
+    override val name: String = "resolved_references"
+    override fun getReferences(psiJavaFile: PsiJavaFile): Sequence<PsiJavaCodeReferenceElement> =
+        psiJavaFile.descendantsOfType<PsiJavaCodeReferenceElement>()
```

Comment:
Interesting way of getting children of a specific type. I knew only by using
```
PsiTreeUtil.collectElementsOfType(root, PsiJavaCodeReferenceElement::class.java)
```
Can you check that both ways provide the same list of nodes